### PR TITLE
CI fixes for R8

### PR DESCRIPTION
**Pull request by ThePotatoGuy**
updates CI script for the following changes:
* lint/distribute with renpy 8.1.1 
* * also means the script installs openGL now
* use updated base MAS zip (its basically no scripts.rpa) - I'll update this later when we have handling for that
* updated version numbers for all actions

This should wait until #9493 is in.

*also I did rebasing in this one to squash the commits. I probably should have separated the changes above into separate commits though - next time.*

### CODE REVIEW

File: .github/workflows/mas_check.yml [hunk 1]
```diff
@@ -22,47 +22,55 @@ jobs:
...
-      #     key: ${{ runner.os }}-mas
-
-      - name: Download base MAS
-      #  if: steps.cache-mas.outputs.cache-hit != 'true'
+      - name: cache base MAS
+        id: cache-mas
+        uses: actions/cache@v4
+        with:
+          path: mas/
+          key: ${{ runner.os }}-mas
```
Comment (Booplicate) [inline .github/workflows/mas_check.yml:62]: Is there a reason to cache MAS per-platform? We always download the same archive. It looks like a dummy cache key that always hits after the first time, maybe we should just use a static string to indicate that? Same for renpy cache.

File: .github/workflows/mas_check.yml [hunk 2]
```diff
@@ -22,47 +22,55 @@ jobs:
...
-      #     path: mas0105/
-      #     key: ${{ runner.os }}-mas
-
-      - name: Download base MAS
-      #  if: steps.cache-mas.outputs.cache-hit != 'true'
+      - name: cache base MAS
+        id: cache-mas
+        uses: actions/cache@v4
+        with:
+          path: mas/
```
Comment (Booplicate) [inline .github/workflows/mas_check.yml:61]:
```suggestion
          path: $MAS_DIR
```

File: .github/workflows/mas_check.yml [hunk 3]
```diff
@@ -22,47 +22,55 @@ jobs:
...
           # Version range or exact version of a Python version to use, using SemVer's version range syntax.
           python-version: 3.10.4 # optional, default is 3.x
 
-      # dl renpy src
+      # get/dl renpy src
+      - name: cache rpy source
+        id: cache-rpy
+        uses: actions/cache@v4
+        with:
+          path: renpy/
```
Comment (Booplicate) [inline .github/workflows/mas_check.yml:44]: Wanna move this to a variable like you did with `MAS_DIR`?

File: .github/workflows/mas_check.yml [hunk 4]
```diff
@@ -72,10 +80,18 @@ jobs:
...
+          sudo apt-get install mesa-utils libglu1-mesa-dev freeglut3-dev mesa-common-dev
+          sudo apt-get install libglew-dev libglfw3-dev libglm-dev
+          sudo apt-get install libao-dev libmpg123-dev
 
       # distribute
       - name: rpy distribute
         run: |
           cd renpy
-          ./renpy.sh launcher distribute "../mas0105/" --package market
+          ./renpy.sh launcher distribute "../$MAS_DIR/" --package market
```
Comment (Booplicate) [inline .github/workflows/mas_check.yml:97]: We should probably make a linux build instead of market here.

**Review by Booplicate — CHANGES_REQUESTED** [review]

File: .github/workflows/mas_check.yml (hunk 1 above)
Comment (ThePotatoGuy) [reply .github/workflows/mas_check.yml]: its there so we we could do building on other platforms, but not necessary now so changed to simple key

File: .github/workflows/mas_check.yml (hunk 2 above)
Comment (ThePotatoGuy) [reply .github/workflows/mas_check.yml]: had to use `${{ env.var }}/` notation to work

File: .github/workflows/mas_check.yml (hunk 3 above)
Comment (ThePotatoGuy) [reply .github/workflows/mas_check.yml]: donezo

File: .github/workflows/mas_check.yml (hunk 4 above)
Comment (ThePotatoGuy) [reply .github/workflows/mas_check.yml]: donezo 2

**Review by multimokia — APPROVED** [review]

**Review by Booplicate — APPROVED** [review]

**Pull request merged**